R/build_stan_data.R

```diff
@@ -0,0 +1,197 @@
+#' Sets normal distribution parameters for sampling
+#'
+#' @param mean Mean of the normal distribution
+#' @param sd Standard deviation of the normal distribution
+#' @return List with specified statistics and name of the model
+#' @export
+sf_normal <- function(mean = 0, sd = 1) {
+  # Restricting normal inputs to be non-negative
+  if (mean < 0 || sd <= 0) {
+    msg <- paste0(
+      "Normal distribution here only accepts",
+      " non-negative values for mean and standard deviation"
+    )
+    message(msg)
+    stop()
+  }
+
+  return(list(mean = mean, sd = sd, name = "normal"))
+}
+
+#' Sets uniform distribution parameters for sampling
+#'
+#' @param min Minimum value of the random variable of the uniform distribution
+#' @param max Maximum value of the random variable of the uniform distribution
+#' @return List with specified statistics and name of the model
+#' @export
+sf_uniform <- function(min = 0, max = 10) {
+  # Restricting uniform inputs to be non-negative
+  if (min < 0 || max < 0) {
+    msg <- paste0(
+      "Uniform distribution here only accepts",
+      " non-negative values for min and max"
+    )
+    message(msg)
+    stop()
+  }
+  if (min >= max) {
+    message("Uniform distribution only accepts min < max")
+  }
+
+  return(list(min = min, max = max, name = "uniform"))
+}
+
+#' Sets empty distribution
+sf_none <- function() {
+  return(list(name = "none"))
+}
+
+#' Generates force-of-infection indexes for heterogeneous age groups
+#'
+#' The max value of the force-of-infection indexes correspond to
+#' the number of foi values to be estimated when sampling.
+#' @inheritParams fit_seromodel
+#' @param group_size Age groups size
+#' @return Integer vector with the indexes numerating each year/age
+#' (depending on the model).
+#' @export
+get_foi_index <- function(
+  serosurvey,
+  group_size
+  ) {
+    checkmate::assert_int(
+      group_size,
+      lower = 1,
+      upper = max(serosurvey$age_max)
+      )
+
+    foi_index <- unlist(
+      purrr::map(
+        seq(
+          1,
+          max(serosurvey$age_max) / group_size,
+          1),
+        rep,
+        times = group_size
+      )
+    )
+
+    foi_index <- append(
```

Comment:
`c()` is simpler and faster in the case where there is not a specific need for `append()`

```suggestion
    foi_index <- c(
```